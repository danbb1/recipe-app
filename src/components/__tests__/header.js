import React from "react"
import { render, screen } from "@testing-library/react"

import Header from "../header"

jest.mock("auth0-js")

describe("Header", () => {
  it("renders correctly without a user", () => {
    render(<Header siteTitle="Recipe App" isAuthenticated={false} user={{}} />)
    screen.getByText("Recipe App")
    screen.getByText("Login")
  })
  it("render correctly with a user", () => {
    render(
      <Header
        siteTitle="Recipe App"
        isAuthenticated
        user={{ nickname: "Test User" }}
      />
    )
    screen.getByText("Logout")
    screen.getByText("T")
  })
})
