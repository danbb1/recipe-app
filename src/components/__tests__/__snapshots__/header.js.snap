// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-3 makeStyles-appBar-10 MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open menu"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h5
        className="MuiTypography-root makeStyles-title-6 MuiTypography-h5"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          
        </a>
      </h5>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Login
        </span>
      </button>
    </div>
  </header>
  <nav>
    <div
      className="PrivateHiddenCss-mdUp-19"
    />
    <div
      className="PrivateHiddenCss-smDown-17"
    >
      <div
        className="MuiDrawer-root MuiDrawer-docked"
      >
        <div
          className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-7 makeStyles-drawerPaper-11 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
        >
          <div>
            <div
              className="makeStyles-toolbar-27"
            />
            <hr
              className="MuiDivider-root"
            />
            <ul
              className="MuiList-root MuiList-padding"
            >
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                href="/"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  className="MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                    />
                  </svg>
                </div>
                <div
                  className="MuiListItemText-root"
                >
                  <span
                    className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Home
                  </span>
                </div>
              </a>
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                href="/recipes/"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  className="MuiListItemText-root MuiListItemText-inset"
                >
                  <span
                    className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Recipes
                  </span>
                </div>
              </a>
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                href="/meal-planner/"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  className="MuiListItemText-root MuiListItemText-inset"
                >
                  <span
                    className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Meal Planner
                  </span>
                </div>
              </a>
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                href="/photos/"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  className="MuiListItemText-root MuiListItemText-inset"
                >
                  <span
                    className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Photos
                  </span>
                </div>
              </a>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
